src/components/course/data/utils.jsx

```diff
@@ -870,3 +873,70 @@ export function transformedCourseMetadata({
   };
   return minimalCourseMetadata;
 }
+
+/**
+ * Takes assignments with the earliestPossibleExpirationDate field and sorts by the
+ * soonest expiring expiration date, along with returns relevant metadata
+ *
+ * @param assignmentObjectArray
+ * @param dateFormat
+ * @returns {
+ *  {
+ *    date: (*|string),
+ *    reason: string,
+ *    sortedExpirationDateData: *,
+ *    soonestExpirationDateData: *
+ *   } |
+ *   {
+ *    date: null,
+ *    reason: null,
+ *    sortedExpirationDateData: null
+ *   } |
+ *   {
+ *    date: null,
+ *    reason: null,
+ *    sortedByExpirationDate: null
+ *   }
+ * }
+ */
+export function getSoonestEarliestPossibleExpirationData({
+  assignments,
+  dateFormat = null,
+}) {
+  if (!assignments?.length) {
+    logInfo(`[sortedByExpirationDate] ${assignments} an empty array`);
```

Comment:
[nit/curious] Is the `[sortedByExpirationDate]` prefix intended to match the function name? E.g., at this point in the code path `sortedByExpirationDate` does not exist (i.e., defined below line 928).

Reply:
nit: given `assignments` is an empty array, i'm not sure how much utility there is in including the variable itself in the `logInfo`?:

<img width="394" alt="image" src="https://github.com/user-attachments/assets/cd11cdfb-7494-4aa3-9490-fb19dcb46a8e">

Would this message make sense as something like, `[sortedByExpirationDate] no assignments provided in array` or similar?

Semi-related (curious for your two cents!), I'm wondering if we might want to treat this and the below `logInfo` as `logError` given it seems in the code path where this function is called, it's only called when there _are_ assigned course runs, we _might_ want to consider it as an error that could trigger an alert for observability/monitoring. For example, if we determine there are assigned course runs, and then pass in empty array or the assignments are all missing `.earliestPossibleExpiration`, would we want to consider that an error? 🤔

Reply:
I had this debate with myself as well when implementing this approach. I guess I overlooked the part where we are deep enough into the code path to expect assignments in the array at this point so a change to `logError` would make the most sense. Changing now 👍🏽 
Also, at one point the function name was `sortedByExpirationDate` but ended up changing the name and not the variable. Updated to take that into account.